pkg/beaconproxy/mongodb.go

```diff
@@ -0,0 +1,142 @@
+package beaconproxy
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/activecm/rita/resources"
+	"github.com/activecm/rita/util"
+	"github.com/globalsign/mgo"
+	"github.com/vbauerster/mpb"
+	"github.com/vbauerster/mpb/decor"
+)
+
+type repo struct {
+	res *resources.Resources
+	min int64
+	max int64
+}
+
+//NewMongoRepository create new repository
+func NewMongoRepository(res *resources.Resources) Repository {
+	min, max, _ := res.MetaDB.GetTSRange(res.DB.GetSelectedDB())
+	return &repo{
+		res: res,
+		min: min,
+		max: max,
+	}
+}
+
+func (r *repo) CreateIndexes() error {
+	session := r.res.DB.Session.Copy()
+	defer session.Close()
+
+	// set collection name
+	collectionName := r.res.Config.T.BeaconProxy.BeaconProxyTable
+
+	// check if collection already exists
+	names, _ := session.DB(r.res.DB.GetSelectedDB()).CollectionNames()
+
+	// if collection exists, we don't need to do anything else
+	for _, name := range names {
+		if name == collectionName {
+			return nil
+		}
+	}
+
+	// set desired indexes
+	indexes := []mgo.Index{
+		{Key: []string{"-score"}},
+		{Key: []string{"src", "src_network_uuid"}},
```

Comment:
Might consider adding an index on "dst" and "dst_network_uuid" so we can easily find connections which passed through a particular proxy server.

Reply:
Added